controllers/array_action/array_mediator_svc.py

```diff
@@ -1321,12 +1438,35 @@ def _start_rcrelationship(self, rcrelationship_id, primary_endpoint_type=None, f
                 logger.warning("failed to start rcrelationship '{}': {}".format(rcrelationship_id, ex))
 
     def create_replication(self, replication_request):
+        if replication_request.replication_type == array_settings.REPLICATION_TYPE_MIRROR:
+            self._create_replication(replication_request)
+        elif replication_request.replication_type == array_settings.REPLICATION_TYPE_EAR:
+            self._create_ear_replication(replication_request)
+
+    def _create_replication(self, replication_request):
         rc_id = self._create_rcrelationship(replication_request.volume_internal_id,
                                             replication_request.other_volume_internal_id,
                                             replication_request.other_system_id,
                                             replication_request.copy_type)
         self._start_rcrelationship(rc_id)
 
+    def _create_ear_replication(self, replication_request):
+        if not self._is_earreplication_supported():
+            logger.info("EAR replication is not supported on the existing storage")
+            return
+
+        replication_policy = replication_request.replication_policy
+
+        # for phase 1 - create empty volume group and move volume into it
```

Comment:
please remove comments
if the code is not clear enough so make it so, if it's only a design so we have a wiki for it.

Reply:
I removed comment and added explanation to design document in wiki